.github/workflows/ci.yml

```diff
@@ -351,6 +372,49 @@ jobs:
           tool: cargo-audit
       - run: cargo audit -D warnings
 
+  generate_coverage:
+    name: "Generate Coverage Artifact"
+    runs-on: ubuntu-latest
+    needs:
+      - cargo_nextest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - type: unit
+          - type: integration
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v3
+        with:
+          name: coverage-profraw
+          path: coverage/profraw
+      - uses: taiki-e/install-action@91af8c38814c3998cb755869e5cbeffd3ab0e462
+        with:
+          tool: cargo-llvm-cov
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          cache: pip
+      - run: pip3 install --user diff-cover
+      - run: for f in coverage/profraw/binaries/*.tar.zst; do tar -x --zstd -f $f; done
+      - run: for f in coverage/profraw/${{matrix.type}}/*.tar.zst; do tar -x --zstd -C target/ -f $f; done
+      - run: cargo llvm-cov show-env | grep -v RUSTFLAGS | tr -d "'" >> "$GITHUB_ENV"
+      - run: mkdir -p coverage/lcov coverage/html
+      - run: cargo llvm-cov report --profile dev-release --lcov --output-path coverage/lcov/${{matrix.type}}.lcov
+      - run: cargo llvm-cov report --profile dev-release --html --hide-instantiations --output-dir coverage/html/${{matrix.type}}
+      - run: git remote -v
+      - run: git fetch origin/master
```

Comment:
Not every PR will be based upon master... Even then it seems somewhat weird to me to compare against the origin/master and not the branch-off point? Like if I added a buncha tests and increased coverage by 10% compared to the branch base, but in master somebody pushed up 100k untested and uncovered lines, comparing against `origin/master` would readily present a decrease in coverage, no?

Reply:
> Even then it seems somewhat weird to me to compare against the origin/master and not the branch-off point?

AFAIU `diff-cover` already defaults to taking the merge-base of the two commits passed in :)

As for PRs against other branches than origin/master, I’m using the same logic as the "protobuf backward compatibility" check here: we basically don’t have these anyway, and when we do we don’t really care about CI failing at these times.